File: 0141. Linked List Cycle.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public bool HasCycle(ListNode head) {
        
        ListNode a = head ;   //一次跳2位
        ListNode b = head ;  //一次跳1位
        
        while(a != null &&  a.next != null ){
            a = a.next.next ;
            b = b.next ;
            
            if (a == b){
                return true ;
            }
        }
            
        return false ;
    }
}

File: 0141. Linked List Cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: ListNode) -> bool:
        
        a = head    #一次跳2位
        b = head    #一次跳1位
        
        while a and a.next:
            a = a.next.next
            b = b.next
            
            if a == b:
                return True
        
        return False